### PR TITLE
doc: use nrfutil in documentation

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
KRKNWK-19513
nrf54 should use nrfutil for flashing

## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: 

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [x] Change has been tested.
- [x] Tests were updated (if applicable).

### CODE REVIEW

**Review by ktaborowski — APPROVED** [review]
lgtm

File: doc/setting_up_sidewalk_environment/setting_up_sidewalk_prototype.rst [hunk 1]
```diff
@@ -86,20 +86,21 @@ The tools required for provisioning are located in the repository (`sdk-nrf`_ an
                --wireless_device_json wireless_device.json
                --device_profile_json device_profile.json
                --addr 0x17c000
+               --output_hex nordic_aws_nrf54l15.hex
 
          .. note::
             The default name of the :file:`mfg.hex` file generated from the :file:`provision.py` script is :file:`nordic_aws_nrf52840.hex`.
-            It is, however, compatible with other supported boards.
+            It is, however, compatible with other supported boards, but this name can be overwritten by parameter `--output_hex` as shown in example.
```
Comment (annwoj) [inline doc/setting_up_sidewalk_environment/setting_up_sidewalk_prototype.rst:93]: If someone runs the command above then the default name won't be a problem anymore, so maybe we should rephrase the entire note to: 

```suggestion
            The ``--output_hex`` parameter is used to specify a custom name for the :file:`mfg.hex` file. 
            This change is optional and does not affect the file's compatibility with other supported boards. 
            If the parameter is not provided, the default name will be :file:`nordic_aws_nrf52840.hex`.
```

+should this be updated for the other tab as well(?)

Comment (RobertGalatNordic) [reply doc/setting_up_sidewalk_environment/setting_up_sidewalk_prototype.rst:93]: ok, change applied

Comment (annwoj) [reply doc/setting_up_sidewalk_environment/setting_up_sidewalk_prototype.rst]: The change is not fully applied - can we edit the entire note as suggested?

**Comment by totyz** [comment]
@annwoj , @wojciechpluszynski - What about compatibility_matrix page (https://jenkins-ncs.nordicsemi.no/job/KRKNG/job/Sidewalk/job/test-sdk-sidewalk/job/master/1316/artifact/html_doc/compatibility_matrix.html ). Do we plan to keep it - if so, it need to be updated

**Review by annwoj — APPROVED** [review]

**Pull request merged**